src/modules/localization/domain/dynamics/constant_acceleration_model.cpp

```diff
@@ -0,0 +1,63 @@
+#include <constant_acceleration_model.hpp>
+
+namespace localization::dynamics {
+
+    ConstantAccelerationModel::ConstantAccelerationModel(double noise_ax, double noise_ay, double noise_atheta) :
+                                                         noise_ax(noise_ax), noise_ay(noise_ay), 
+                                                         noise_atheta(noise_atheta) {}
+
+    void ConstantAccelerationModel::set_matrix_values(Eigen::VectorXd &x, Eigen::MatrixXd &F, Eigen::MatrixXd &P, Eigen::MatrixXd &Q){
+    // Set Matrix Values
+
+        x = Eigen::VectorXd(9);
+          // state covariance matrix P
+        P = Eigen::MatrixXd(9, 9);
+        P <<  100, 0, 0, 0, 0, 0, 0, 0, 0, 
+              0, 100, 0, 0, 0, 0, 0, 0, 0,
+              0, 0, 100, 0, 0, 0, 0, 0, 0,
+              0, 0, 0, 100, 0, 0, 0, 0, 0,
+              0, 0, 0, 0, 100, 0, 0, 0, 0,
+              0, 0, 0, 0, 0, 100, 0, 0, 0,
+              0, 0, 0, 0, 0, 0, 100, 0, 0,
+              0, 0, 0, 0, 0, 0, 0, 100, 0,
+              0, 0, 0, 0, 0, 0, 0, 0, 100;
+
+        // the initial transition matrix F_
+        double dt = 0;
+        F = Eigen::MatrixXd(9, 9);
+        update_state_matrix(F, dt);
+        Q = Eigen::MatrixXd(9, 9);
+        update_process_noise_matrix(Q, dt);
+     }
+
+     void ConstantAccelerationModel::update_process_noise_matrix(Eigen::MatrixXd& Q, double dt) {
+        double dt_2 = dt*dt;
+        double dt_3 = dt_2*dt;
+        double dt_4 = dt_3*dt;
+
+        Q<< dt_4*noise_ax/4, 0,               0,                   dt_3*noise_ax/2, 0,               0,                   dt_2*noise_ax/2, 0,               0,
+             0,               dt_4*noise_ay/4, 0,                   0,               dt_3*noise_ay/2, 0,                   0,               dt_2*noise_ay/2, 0,
+             0,               0,               dt_4*noise_atheta/4, 0,               0,               dt_3*noise_atheta/2, 0,               0,               dt_2*noise_atheta/2,
+             dt_3*noise_ax/2, 0,               0,                   dt_2 * noise_ax, 0,               0,                   dt*noise_ax,     0,               0,
+             0,               dt_3*noise_ay/2, 0,                   0,               dt_2*noise_ay,   0,                   0,               dt*noise_ay,     0,
+             0,               0,               dt_3*noise_atheta/2, 0,               0,               dt_2*noise_atheta,   0,               0,               dt*noise_atheta,
+             dt_2*noise_ax/2, 0,               0,                   dt*noise_ax,     0,               0,                   noise_ax,        0,               0,
```

Comment:
Should there be a division by 2 in element (6,1) which is dt_2*noise_ax/2 ?

Reply:
That is correct. But the other two are wrong. Good catch